frontend/src/pages/workspace/Index.tsx

```diff
@@ -25,18 +23,23 @@ import { Document } from "../../hooks/api/types/document.d";
 import InfiniteScroll from "react-infinite-scroller";
 import CreateModal from "../../components/modals/CreateModal";
 import AddIcon from "@mui/icons-material/Add";
+import BoardTab from "../../components/workspace/BoardTab";
+import TableTab from "../../components/workspace/TableTab";
 
-const tabs = [
-	"BOARD",
-	// "TABLE"
-];
+const TABS = ["BOARD", "TABLE"] as const;
+type TabType = (typeof TABS)[keyof typeof TABS];
```

Comment:
This type will work with `object`, but not be inferred as intended.

Since ⁠`TABS` is an ⁠`Array<string>`, this type will be inferred with all the properties that an ⁠`Array<string>` has.

```ts
type TabType = (typeof TABS)[keyof typeof TABS];
/*
"BOARD" | "TABLE" | 2 | (() => ArrayIterator<"BOARD" | "TABLE">) | {
    concat?: boolean | undefined;
    join?: boolean | undefined;
    // …
}
*/
```

In such cases, we can use an [Indexed Access Type](https://www.typescriptlang.org/docs/handbook/2/indexed-access-types.html) to derive a union type representing the array's elements.

```suggestion
type TabType = typeof TABS[number];
```